src/core/thread/mle.cpp

```diff
@@ -128,14 +128,6 @@ Mle::Mle(Instance &aInstance)
 
     mLeaderAloc.InitAsThreadOriginRealmLocalScope();
 
-#if OPENTHREAD_CONFIG_TMF_NETDATA_SERVICE_ENABLE
-    for (Ip6::Netif::UnicastAddress &serviceAloc : mServiceAlocs)
-    {
-        serviceAloc.InitAsThreadOriginRealmLocalScope();
-        serviceAloc.GetAddress().GetIid().SetLocator(Mac::kShortAddrInvalid);
-    }
-#endif
-
```

Comment:
_A note to help with review_: This is now done by the `ServiceAloc` constructor initializing it.